File: python/program/Vehicle.py
class Vehicle:
    def __init__(self, plat_nomor, merk, tahun_produksi, warna):
        self._plat_nomor = plat_nomor
        self._merk = merk
        self._tahun_produksi = tahun_produksi
        self._warna = warna

    def get_plat_nomor(self):
        return self._plat_nomor

    def get_merk(self):
        return self._merk

    def get_tahun_produksi(self):
        return self._tahun_produksi

    def get_warna(self):
        return self._warna

    def set_plat_nomor(self, plat_nomor):
        self._plat_nomor = plat_nomor

    def set_merk(self, merk):
        self._merk = merk

    def set_tahun_produksi(self, tahun_produksi):
        self._tahun_produksi = tahun_produksi

    def set_warna(self, warna):
        self._warna = warna

    def print_data(self):
        print(f"Plat Nomor: {self._plat_nomor}")
        print(f"Merk: {self._merk}")
        print(f"Tahun Produksi: {self._tahun_produksi}")
        print(f"Warna: {self._warna}")

File: cpp/program/Vehicle.cpp
#include "Vehicle.h"
#include <iostream>

Vehicle::Vehicle(const std::string &platNomor, const std::string &merk, int tahunProduksi, const std::string &warna)
    : platNomor(platNomor), merk(merk), tahunProduksi(tahunProduksi), warna(warna) {}

std::string Vehicle::getPlatNomor() const {
    return platNomor;
}

void Vehicle::setPlatNomor(const std::string &value) {
    platNomor = value;
}

std::string Vehicle::getMerk() const {
    return merk;
}

void Vehicle::setMerk(const std::string &value) {
    merk = value;
}

int Vehicle::getTahunProduksi() const {
    return tahunProduksi;
}

void Vehicle::setTahunProduksi(int value) {
    tahunProduksi = value;
}

std::string Vehicle::getWarna() const {
    return warna;
}

void Vehicle::setWarna(const std::string &value) {
    warna = value;
}

void Vehicle::printData() const {
    std::cout << "Plat Nomor: " << platNomor << std::endl;
    std::cout << "Merk: " << merk << std::endl;
    std::cout << "Tahun Produksi: " << tahunProduksi << std::endl;
    std::cout << "Warna: " << warna << std::endl;
}
